modules/imgproc/src/hough.cpp

```diff
@@ -1612,6 +1612,141 @@ void HoughCircles( InputArray _image, OutputArray _circles,
+    {
+        for (int deg = deg_min; deg < deg_max; deg += deg_step)
+        {
+            // Calc rho
+            rad = TO_RAD(deg);
+            rho = (double)point.at(i).x * cos(rad) + (double)point.at(i).y * sin(rad);
+            // Calc index
+            rho_index = (int)((rho - rho_min) / rho_step);
+            theta_index = (int)((double)(deg - deg_min) / (double)deg_step);
+
+            if (((0 <= rho_index) && (rho_index < rho_size)) &&
+                ((0 <= theta_index) && (theta_index < theta_size)))
+            {
+                if (*(plane + (rho_index * theta_size) + theta_index) < INT_MAX)
+                {
+                    *(plane + (rho_index * theta_size) + theta_index) += 1;
+                }
+            }
+            else
+            {
+                // Do nothing.
+            }
+        }
+    }
+}
+
+static int SelectHoughLines( const int *plane,
+                             double rho_min, double rho_max, double rho_step,
+                             double theta_min, double theta_max, double theta_step,
+                             std::vector<Vec3d>& lines )
+{
+    int ret = 0, cnt = 0;
+    int votes = 0, max_votes = 0;
+    int rho_size = (int)((rho_max - rho_min) / rho_step);
+    int theta_size = (int)((TO_DEG(theta_max) - TO_DEG(theta_min)) / TO_DEG(theta_step));
+
+    for( int rho = 0; rho < rho_size; rho++ )
+    {
+        for( int theta = 0; theta < theta_size; theta++ )
+        {
+            votes = *(plane + (rho * theta_size) + theta);
+            if( votes >= max_votes )
+            {
+                lines[cnt] = Vec3d((double)votes, ((double)rho * rho_step + rho_min), ((double)theta * theta_step + theta_min));
+                max_votes = votes;
+                ret = cnt;
+                cnt++ ;
+                if( cnt >= (int)lines.size() ){ cnt = 0; }
+            }
+            else
+            {
+                // Do nothing.
+            }
+        }
+    }
+    return ret;
+}
+
+int HoughLinesUsingSetOfPoints( InputArray _point, OutputArray _lines, int lines_max,
+                                double rho_min, double rho_max, double rho_step,
+                                double theta_min, double theta_max, double theta_step )
+{
+    int max_polar_index = 0;
+
+    if( (_point.type() == CV_32FC2) || (_point.type() == CV_32SC2) )
+    {
+        std::vector<Point2f> point;
+        _point.copyTo(point);
+        if( lines_max > 0 )
+        {
+            std::vector<Vec3d> lines(lines_max, 0.0f);
+
+            if( ((rho_max   - rho_min)   > 0) &&
+                ((theta_max - theta_min) > 0) &&
+                ((rho_step   > 0))            &&
+                ((theta_step > 0)) )
+            {
+                int rho_size = (int)((rho_max - rho_min) / rho_step);
+                int theta_size = (int)((TO_DEG(theta_max) - TO_DEG(theta_min)) / TO_DEG(theta_step));
+
+                int *plane = (int*)fastMalloc(sizeof(int) * rho_size * theta_size);
```

Comment:
Rounding to zero can occur for `rho_size` and `theta_size`, need to check this before allocating memory.
Also it's better to use `AutoBuffer<int>` for memory allocation instead of malloc/free functions
(or even any kind of Mat to fill the data with zeros and avoid using `memset`)

Reply:
Thank you for a review.

>t'd be better to reuse the parts of code from HoughLinesStandard from hough.cpp.
->I think so, too. Compared to my code, the HoughLinesStandard function is refined and proven. I reuse existing code and fix it.I should have reused the code from the beginning.

Reply:
I fixed the code by reused the parts of code from HoughLinesStandard from hough.cpp.
Could you check this code when you have time?I know you are very busy, but I would appreciate hearing from you.